test/bin/invalid-package-version.t/run.t

```diff
@@ -0,0 +1,55 @@
+We want to make sure the error messages are sensible, and as such if the user
+picks a version that doesn't exist we want to make them aware of it.
+
+We setup the default base repository
+
+  $ gen-minimal-repo
+
+Here we define a package test that depends on a package `a`:
+
+  $ opam show --no-lint --raw -fdepends ./existing.opam
+  "dune" "a"
+
+We have a local repo that defines a package `a` that satisfies the predicate,
+with a version that is valid and can be picked.
+
+  $ cat repo/packages/a/a.0.1/opam > /dev/null
+
+opam-monorepo solver should successfully pick a.0.1:
+
+  $ opam-monorepo lock existing
+  ==> Using 1 locally scanned package as the target.
+  ==> Found 9 opam dependencies for the target package.
+  ==> Querying opam database for their metadata and Dune compatibility.
+  ==> Calculating exact pins for each of them.
+  ==> Wrote lockfile with 1 entries to $TESTCASE_ROOT/existing.opam.locked. You can now run opam monorepo pull to fetch their sources.
+  $ cat existing.opam.locked | grep "\"a\"\s\+{"
+    "a" {= "0.1" & vendor}
+
+Yet if we attempt to use the same package, but pick a version that doesn't
+exist in our repo:
+
+  $ opam show --no-lint --raw -fdepends ./toonew.opam
+  "dune" "a" {>= "1.0"}
+
+opam-monorepo should fail with some error code and display an error message
+that there is no version of `a` that matches the constraint.
+
+(grep appends a NUL byte at the end, hence the head call, this is not important
+to the test)
+
+  $ opam-monorepo lock toonew 2> errors
+  ==> Using 1 locally scanned package as the target.
+  [1]
+  $ grep -Pazo "(?s)opam-monorepo: \[ERROR\].*(?=opam-monorepo)" < errors | head --bytes=-1
+  opam-monorepo: [ERROR] There is no eligible version of a that matches >= 1.0
+
+We should also produce the right error message with all the constraints when we have multiple constaints
+
+  $ opam show --no-lint --raw -fdepends ./multiple-constraint.opam
+  "dune" "a" {>= "1.1" & < "2.0"}
```

Comment:
I think this won't be testing the right thing. To test the initial problem, the constraint have to come from different packages I think. This can be done by introducing another package, itself depending on a!

Reply:
After offline discussions we concluded that there's no easy way to test this scenario and we can improve this part of the testing as we start working on the generic solver error messages.